client-core/src/service/key_service.rs

```diff
@@ -56,7 +92,152 @@ where
 
     /// Clears all storage
     pub fn clear(&self) -> Result<()> {
-        self.storage.clear(KEYSPACE)
+        self.storage.clear(KEYSPACE).expect("clear keyspace");
+        self.storage
+            .clear(KEYSPACE_HD)
+            .expect("clear keyspace for hd");
+        Ok(())
+    }
+
+    /// get random mnemonic
+    pub fn get_random_mnemonic(&self) -> String {
+        let mnemonic = Mnemonic::new(MnemonicType::Words24, Language::English);
+        mnemonic.clone().to_string()
+    }
+
+    /// is hd
+    pub fn is_hd_wallet(&self, name: &str, passphrase: &SecUtf8) -> bool {
+        let key = name.as_bytes();
+        let value = self.read_value(passphrase, &key[..]);
+        value.is_some()
+    }
+    /// generate seed from mnemonic
+    pub fn generate_seed(&self, mnemonic: &str, name: &str, passphrase: &SecUtf8) -> Result<()> {
+        debug!("hdwallet generate seed={}", mnemonic);
+        let mnemonic = Mnemonic::from_phrase(&mnemonic.to_string(), Language::English).unwrap();
+        let seed = Seed::new(&mnemonic, "");
+        self.storage
+            .set_secure(KEYSPACE_HD, name, seed.as_bytes().into(), passphrase)?;
```

Comment:
Just passing `seed` or `&seed` should work.
```suggestion
            .set_secure(KEYSPACE_HD, name, seed, passphrase)?;
```